src/Type/ParserNodeTypeToPHPStanType.php

```diff
@@ -2,17 +2,89 @@
 
 namespace PHPStan\Type;
 
+use PhpParser\Node\ComplexType;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\BinaryOp\BitwiseOr;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\UnaryMinus;
 use PhpParser\Node\Identifier;
 use PhpParser\Node\Name;
 use PhpParser\Node\NullableType;
+use PhpParser\Node\Scalar\DNumber;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\Node\Scalar\String_;
 use PHPStan\Reflection\ClassReflection;
 use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\Constant\ConstantIntegerType;
 
 class ParserNodeTypeToPHPStanType
 {
 
+	public static function resolveParameterDefaultType(?Expr $type): Type
+	{
+		if ($type instanceof ConstFetch) {
+			return self::resolve(new Identifier($type->name->toString()), null);
+		} elseif ($type instanceof ClassConstFetch) {
+			if ($type->name instanceof Identifier) {
+				$constantName = $type->name->name;
+				if (!($type->class instanceof Name)) {
+					throw new \PHPStan\ShouldNotHappenException();
+				}
+
+				$constantClass = (string) $type->class;
+				$constantClassType = new ObjectType($constantClass);
+
+				if (!$constantClassType->hasConstant($constantName)->yes()) {
+					return new ErrorType();
+				}
+
+				return $constantClassType->getConstant($constantName)->getValueType();
+			}
+
+			throw new \PHPStan\ShouldNotHappenException();
+		} elseif ($type instanceof String_ || $type instanceof LNumber || $type instanceof DNumber) {
+			return ConstantTypeHelper::getTypeFromValue($type->value);
+		} elseif ($type instanceof Array_) {
+			return ConstantTypeHelper::getTypeFromValue($type->items);
```

Comment:
This doesn't seem right. ConstantTypeHelper will not resolve ArrayItem instances correctly.

Reply:
fixed